Description: Add a pkg-config file for the shared libraries.
Author: Benjamin Drung <bdrung@debian.org>

--- a/live/Makefile.head
+++ b/live/Makefile.head
@@ -1 +1,4 @@
+PREFIX = /usr/local
+LIBDIR = /usr/local/lib
+VERSION = $(shell grep LIVEMEDIA_LIBRARY_VERSION_STRING liveMedia/include/liveMedia_version.hh | sed 's/.*"\([^"]*\)".*/\1/')
 ##### Change the following for your environment:
--- /dev/null
+++ b/live/live555.pc.in
@@ -0,0 +1,9 @@
+prefix=@PREFIX@
+libdir=@LIBDIR@
+includedir=${prefix}/include
+
+Name: live555
+Description: multimedia RTSP streaming library
+Version: @VERSION@
+Cflags: -I${includedir}/liveMedia -I${includedir}/groupsock -I${includedir}/BasicUsageEnvironment -I${includedir}/UsageEnvironment
+Libs: -L${libdir} -lliveMedia -lgroupsock -lBasicUsageEnvironment -lUsageEnvironment
--- live-2021.04.06/live/Makefile.tail.orig	2021-04-29 22:32:18.998971852 +0200
+++ live-2021.04.06/live/Makefile.tail	2021-04-30 06:53:31.459794869 +0200
@@ -25,7 +25,12 @@ all:
 	@echo
 	@echo "For more information about this source code (including your obligations under the LGPL), please see our FAQ at http://live555.com/liveMedia/faq.html"
 
-install:
+install_shared_libraries:
+	install -d $(DESTDIR)$(LIBDIR)/pkgconfig
+	sed "s#@PREFIX@#$(PREFIX)#;s#@LIBDIR@#$(LIBDIR)#;s#@VERSION@#$(VERSION)#" live555.pc.in > $(DESTDIR)$(LIBDIR)/pkgconfig/live555.pc
+	chmod 644 $(DESTDIR)$(LIBDIR)/pkgconfig/live555.pc
+
+install: $(INSTALL2)
 	cd $(LIVEMEDIA_DIR) ; $(MAKE) install
 	cd $(GROUPSOCK_DIR) ; $(MAKE) install
 	cd $(USAGE_ENVIRONMENT_DIR) ; $(MAKE) install
